pkg/pillar/vault/handler_zfs.go

```diff
@@ -83,6 +90,20 @@ func (h *ZFSHandler) RemoveDefaultVault() error {
 // SetupDefaultVault setups vaults on zfs, using zfs native encryption support
 func (h *ZFSHandler) SetupDefaultVault() error {
 	if !etpm.IsTpmEnabled() {
+		if base.IsHVTypeKube() {
+			if zfs.DatasetExist(h.log, types.SealedDataset) {
+				return MountVaultZvol(h.log, types.SealedDataset)
```

Comment:
I find the lack of ext4 in the name of the functions a potential source of confusion.
This function does an ext4 mount of the zvol which is the vault.